src/button.js

```diff
@@ -32,11 +41,16 @@ const Button = ({ children, isLoading, noSpacing, type, ...props }) => {
   }
 
   return (
-    <MUIButton className={clsx({ [classes.label]: isLoading,
-      [classes.noSpacing]: noSpacing,
-    })} data-testid='button' {...styleProps} {...props} >
-      {children}
-    </MUIButton>
+    <div>
+      <MUIButton className={clsx({
```

Comment:
We can probably rewrite this part. It is currently divided by style vs non-styles due to some flaws below.

1) `type` in `styleProps` is not native MUI props so there is no point in passing it down.
2) `variant` is inherited from `...props` which expose itself at the abstracted level of button. This might result in an unwanted style effect when the variant prop is configured externally. 

With the prop types issue, one way to work around this would be declaring variant on the JSX side then dynamically manipulate it. 

```jsx
const whichVariant = type === 'primary' ? 'contained' : type === 'secondary' ? 'outlined' : 'text' 
(...)
  <MUIButton variant={whichVariant} {...props} >
        {children}
      </MUIButton>

Reply:
how about getting rid of type. Changing type to variant and then having a checkpoint if it is neither of the 3 options makes 'contained' ? will push this use case please gimme your opinion about it

Reply:
I don't mind styling through `variant` at the `Button` level, but removing `type` could cause some backward compatibility issue. I think the safest option would be not modifying its current attributes (adding new attributes are fine from this issue), but if you want to go this way, we can keep `type` until the Button components in other projects are updated at least.